File: frontend/src/types/class.ts
export interface Class {
  id: number;
  name: string;
  grade?: string | null;
  description?: string | null;
  created_at: string;
}

export interface ClassCreate {
  name: string;
  grade?: string;
  description?: string;
}

export interface ClassUpdate {
  name?: string;
  grade?: string;
  description?: string;
}

File: backend/app/schemas/class_.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class ClassCreate(BaseModel):
    name: str
    grade: Optional[str] = None
    description: Optional[str] = None


class ClassUpdate(BaseModel):
    name: Optional[str] = None
    grade: Optional[str] = None
    description: Optional[str] = None


class ClassResponse(BaseModel):
    id: int
    name: str
    grade: Optional[str] = None
    description: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}
